factories/resolverFactory.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+/*jslint latedef:false*/
+
+(function() {
+  angular.module('o19s.splainer-search')
+    .factory('ResolverFactory', [
+      '$q',
+      'searchSvc',
+      'solrUrlSvc',
+      'normalDocsSvc',
+      ResolverFactory
+    ]);
+
+  function ResolverFactory($q, searchSvc, solrUrlSvc, normalDocsSvc) {
+    var Resolver = function(ids, settings, chunkSize) {
+      var self        = this;
+
+      self.settings   = settings;
+      self.ids        = ids;
+      self.docs       = [];
+      self.args       = {};
+      self.config     = {};
+      self.queryText  = null;
+      self.fieldSpec  = self.settings.createFieldSpec();
+      self.chunkSize  = chunkSize;
+
+      self.fetchDocs  = fetchDocs;
+
+      if ( self.settings.searchEngine === undefined || self.settings.searchEngine === 'solr' ) {
+        var escapeIds = function(ids) {
+          var newIds = [];
+          angular.forEach(ids, function(id) {
+            newIds.push(solrUrlSvc.escapeUserQuery(id));
+          });
+          return newIds;
+        };
+
+        var allIdsLuceneQuery = self.fieldSpec.id + ':(';
+        allIdsLuceneQuery += escapeIds(ids).join(' OR ');
+        allIdsLuceneQuery += ')';
+        self.queryText = allIdsLuceneQuery;
+
+        self.args = {
+          defType: ['lucene'],
+          rows: [ids.length],
+          q: ['#$query##']
+        };
+      } else if ( settings.searchEngine === 'es' ) {
+        self.args = {
+          'query': {
+            'ids': {
```

Comment:
nice! Just want to make sure this isn't cutting edge though. Is this a reasonably old enough query feature that most Elasticsearch instances would have the ids query?